pkg/controller/splunkforwarder/splunkforwarder_controller.go

```diff
@@ -0,0 +1,173 @@
+	// Watch for changes to secondary resource Pods and requeue the owner SplunkForwarder
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &sfv1alpha1.SplunkForwarder{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileSplunkForwarder implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileSplunkForwarder{}
+
+// ReconcileSplunkForwarder reconciles a SplunkForwarder object
+type ReconcileSplunkForwarder struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a SplunkForwarder object and makes changes based on the state read
+// and what is in the SplunkForwarder.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileSplunkForwarder) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling SplunkForwarder")
+
+	// Fetch the SplunkForwarder instance
+	instance := &sfv1alpha1.SplunkForwarder{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	{ // See if our Secret exists
+		secFound := &corev1.Secret{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: config.SplunkAuthSecretName, Namespace: request.Namespace}, secFound)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	var updateDaemonSet bool = false
+	{ // ConfigMaps
+		// Define a new ConfigMap object
+		configMaps := kube.GenerateConfigMaps(instance.Spec.SplunkInputs, request.NamespacedName)
+
+		// Define it outside the loop
+		cmFound := &corev1.ConfigMap{}
+
+		for _, configmap := range configMaps {
+			// Set SplunkForwarder instance as the owner and controller
+			if err := controllerutil.SetControllerReference(instance, configmap, r.scheme); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			// Check if this ConfigMap already exists
+			cmFound = &corev1.ConfigMap{} // reset cmFound
+			err = r.client.Get(context.TODO(), types.NamespacedName{Name: configmap.Name, Namespace: configmap.Namespace}, cmFound)
+			if err != nil && errors.IsNotFound(err) {
```

Comment:
We have been using switch statements for this instead of `else if` https://github.com/openshift/aws-account-operator/blob/master/pkg/controller/account/account_controller.go#L480-L495

Reply:
In your example you are using your own custom errors, how would I do this with the built in client errors?

Reply:
Talked with James, seems we use different methods between different operators, should have a larger team discussion around if we want to make one way or another standard. Resolving for now